Expose REST API modules

ya7on opened the pull request:
**PR summary**

This modification make REST API modules public. Without this changes it is impossible to use any internal types. E.g. it is impossible to use `Transaction` type as function argument, because it cannot be imported.

ya7on committed:
Expose REST API modules

liketurbo commented:
Hi, sorry it took me so long to reply. I don’t think there’s a function that accepts Transaction as an argument. Maybe there’s something I’m missing—could you provide a use case?

ya7on commented:
Sorry, maybe i was not clear enough. I mean this crate types are not accessible from outside. Models cannot be imported from this crate, they are private.

Example of use case when it will be helpful, but at the moment it's impossible:
```rust
async fn traverse_trace_childs(trace: Trace) { // Trace struct cannot be imported and used here, its private
  todo!()
}

///

let ton_client = RestApiClientV2::new(Network::Testnet, None);
let trace = ton_client.get_trace("TRACE").await?;
traverse_trace_childs(trace).await;
```

I mentioned Transaction just as an example. Any other types from rest_api module are not public too.

When i used this crate and tried to get info about transaction then pass returned value to another function i met with issue that i can't import returning type and use it as argument type because it is private

liketurbo commented:
I see now. In that case, exposing `models` does make sense, and so does exposing `TonApiError` from the `errors` module. For the rest, I don’t see a use case — please correct me if I’m wrong. Could you update it? And it’s ready to be merged 🏁

ya7on committed:
Fix REST API modules that should be public

reverted changes from previous commit, set public only `models` submodule and reexported error structure

ya7on commented:
I fixed it. Now only one `models` module is public and i reexported error structure. Is it OK now?

liketurbo commented:
Looks great! Merging now. Thanks 🙏
